### DISCUSSION TITLE
How come rootless containers don't require packet forwarding on the host?

### DISCUSSION THREAD

**Discussion by adrian5** [containers/podman#17138] [Q&A] [answered]
I noticed that with podman, rootless containers have internet connectivity without enabling the host's packet forwarding:

```console
$ sysctl net.ipv4.ip_forward
net.ipv4.ip_forward = 0
```

```console
$ nft list ruleset
chain Fwd {
    type filter hook forward priority filter; policy drop;
}
```

Is this because the TAP interface the containers use is (so to speak) one level below the IP/packet layer, bypassing that need?

**Comment by mheon** [comment, marked as answer]
I wouldn't say it operates below the IP/packet layer - more that it manually replicates the NAT/forwarding performed by the kernel in root containers.

**Reply by adrian5** [reply]
I see. Documentation on tun/tap seems sparse and articles mostly cover the use of bridge interfaces which I don't think are involved here. But delving further into this risks veering off-topic, so thanks!
